sw/device/silicon_creator/manuf/skus/earlgrey_a0/sival_bringup/ft_personalize_3.c

```diff
@@ -138,7 +139,19 @@ static status_t personalize(ujson_t *uj) {
   LOG_INFO("Exporting certificates ...");
   RESP_OK(ujson_serialize_manuf_cert_perso_data_out_t, uj, &out_data);
 
-  // TODO(#19455): Load endorsed UDS certificate and write it to flash.
+  // Import endorsed certificates from the provisioning appliance.
+  LOG_INFO("Importing certificates ...");
+  TRY(ujson_deserialize_manuf_endorsed_certs_t(uj, &endorsed_certs));
+
+  // Write the endorsed UDS certificate to flash and ack to host.
+  TRY(flash_ctrl_info_erase(&kFlashCtrlInfoPageUdsCertificate,
+                            kFlashCtrlEraseTypePage));
+  TRY(flash_ctrl_info_write(
```

Comment:
Should the certificate size be written somewhere? I don't think this writes it to the flash

Reply:
This is encoded in the ASN.1, no? Ultimately we will need functions to read out the ASN.1 from flash to extract various values. But for now I think we can just write the bytes to the page. When we read out the bytes, we can just read the maximum number of cert bytes from the flash info page, b/c each cert is isolated to its own page currently. We may be some zeros at the end, but that is ok.